tests/cmsis_rtos_v1/src/signal.c

```diff
@@ -69,21 +111,91 @@ void test_signal_events_timeout(void)
 	 */
 	evt = osSignalWait(SIGNAL1, TIMEOUT);
 	zassert_equal(evt.status, osEventTimeout, "");
+	osThreadTerminate(id1);
 }
 
 void test_signal_events_signalled(void)
 {
-	osThreadId id1, id2;
+	osThreadId id1, id2, id3;
 	osEvent evt;
+	int signals;
 
 	id1 = osThreadCreate(osThread(Thread_1), osThreadGetId());
 	zassert_true(id1 != NULL, "Thread creation failed");
 
 	id2 = osThreadCreate(osThread(Thread_2), osThreadGetId());
 	zassert_true(id2 != NULL, "Thread creation failed");
 
-	/* wait for a signal */
+	/* wait for multiple signals */
 	evt = osSignalWait(SIGNAL, TIMEOUT);
 	zassert_equal(evt.status, osEventSignal, "");
 	zassert_equal((evt.value.signals & SIGNAL), SIGNAL, "");
+
+	signals = osSignalClear(osThreadGetId(), SIGNAL);
+	zassert_not_equal(signals, 0x80000000, "");
+
+	/* Set any single signal */
+	signals = osSignalSet(osThreadGetId(), SIGNAL1);
+	zassert_not_equal(signals, 0x80000000, "");
+
+	/* wait for any single signal flag */
+	evt = osSignalWait(0, TIMEOUT);
+	zassert_equal(evt.status, osEventSignal, "");
+	zassert_equal(evt.value.signals, SIGNAL1, "");
+
+	/* validate by passing invalid parameters */
+	zassert_equal(osSignalSet(NULL, 0), 0x80000000,
+				  "NULL signal set unexpectedly");
+	zassert_equal(osSignalClear(NULL, 0), 0x80000000,
+				  "NULL signal cleared unexpectedly");
+	/* cannot wait for Flag mask with MSB set */
+	zassert_equal(osSignalWait((int32_t)0x80000000, 0).status, osErrorValue,
+				  "signal wait passed unexpectedly");
+	zassert_equal(osSignalSet(osThreadGetId(), (int32_t)0x80000000),
+				  0x80000000, "signal set unexpectedly");
+
+	id3 = osThreadCreate(osThread(Thread_3), osThreadGetId());
+	zassert_true(id3 != NULL, "Thread creation failed");
+
+	/* Let id3 run to trigger SIGNAL_FLAG */
+	osDelay(10);
+
+	osThreadTerminate(id1);
+	osThreadTerminate(id2);
+	osThreadTerminate(id3);
+}
+
+/* IRQ offload function handler to set signal flag */
+static void offload_function(void *param)
+{
+	u32_t tid = (u32_t)param;
+	int signals;
+
+	/* Make sure we're in IRQ context */
+	zassert_true(_is_in_isr(), "Not in IRQ context!");
+
+	signals = osSignalSet((osThreadId)tid, ISR_SIGNAL);
+	zassert_not_equal(signals, 0x80000000, "");
+}
+
+void test_signal_from_isr(void const *thread_id)
+{
+	/**TESTPOINT: Offload to IRQ context*/
+	irq_offload(offload_function, (void *)thread_id);
+}
+
+osThreadDef(test_signal_from_isr, osPriorityHigh, 3, 0);
+
+void test_signal_events_isr(void)
+{
+	osThreadId id;
+	osEvent evt;
+
+	id = osThreadCreate(osThread(test_signal_from_isr), osThreadGetId());
+	zassert_true(id != NULL, "Thread creation failed");
+	osDelay(10);
+	evt = osSignalWait(ISR_SIGNAL, TIMEOUT);
+	zassert_equal(evt.status, osEventSignal, "");
```

Comment:
Whether we are asserting for either positive or negative cases, it would be good to have the print statement with the zassert, instead of `NULL`, same with the other checks here..

Reply:
makes sense, addressed. Thanks.